### DISCUSSION TITLE
Having Svn and Git Share The Same Sandbox?

### DISCUSSION THREAD

**Discussion by dtashley** [community/community#22880] [New to GitHub]
<p>I’m migrating from Svn to Git, and during the migration process, I’d like to have Svn and Git share a sandbox.  What I mean is that my sandbox would have both a .svn and .git folder.  My workflow would be to make code changes, commit to Svn, then commit to Git.</p>
<p>Question <span class="hashtag">#1:</span>  Has anyone done this?  Will it work?  Can the two products share a sandbox?</p>
<p>The reason the migration will take a bit of time is that I use Svn keyword substitution to embed strings that are used by the software to figure out what version it is.  There doesn’t seem to be the equivalent of keyword substitution in Git.</p>
<p>Question <span class="hashtag">#2:</span>  Are there any example smudge and clean scripts anywhere on Github?</p>
<p>Thanks!</p>

**Comment by sfynx** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/dtashley">@dtashley</a> wrote:</p>
</blockquote>
<blockquote>
<p>I’m migrating from Svn to Git, and during the migration process, I’d like to have Svn and Git share a sandbox.  What I mean is that my sandbox would have both a .svn and .git folder.  My workflow would be to make code changes, commit to Svn, then commit to Git.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Question <span class="hashtag">#1:</span>  Has anyone done this?  Will it work?  Can the two products share a sandbox?</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>Here it could be a challenge to keep the ignored files/directories in sync for both systems (except for each other’s .svn and .git directories) so you do not add anything unintentionally, and preventing the two remote repositories from diverging. In my experience the <a href="https://git-scm.com/docs/git-svn" rel="nofollow noopener">git-svn</a> mechanism is very handy to assist in such a migration, including pushing any outstanding changes to both remotes and converting ignore lists.</p>
